src/components/organisms/topbar/dashboard.tsx

```diff
@@ -39,23 +52,141 @@ export const DashboardTopbar = () => {
 		});
 	};
 
+	useEffect(() => {
+		const getAndSaveFiles = async () => {
+			if (!projectId || !templateFiles) return;
+
+			const fileEntries = Object.entries(templateFiles).map(([path, fileNode]) => {
+				const content = "content" in fileNode ? fileNode.content : "";
+
+				return [path, new Uint8Array(new TextEncoder().encode(content))];
+			});
+
+			await saveAllFiles(Object.fromEntries(fileEntries));
+			addToast({
+				message: t("projectCreatedSuccessfully"),
+				type: "success",
+			});
+
+			getProjectsList();
+			navigate(`/projects/${projectId}`);
+		};
+
+		getAndSaveFiles();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [projectId]);
+
+	const handleImportFile = async (event: React.ChangeEvent<HTMLInputElement>) => {
+		const file = event.target.files?.[0];
+		if (!file) return;
+		setLoadingImportFile(true);
+
+		try {
+			const zip = await JSZip.loadAsync(file);
+			const fileStructure = await processZipContent(zip);
+
+			const manifestFileNode = fileStructure["autokitteh.yaml"];
+			const manifestContent = manifestFileNode && "content" in manifestFileNode ? manifestFileNode.content : null;
+			if (!manifestContent) {
+				addToast({
+					message: tTemplates("projectCreationFailed"),
+					type: "error",
+				});
+
+				LoggerService.error(
+					namespaces.manifestService,
+					`${tTemplates("projectCreationFailedExtended", { error: tTemplates("projectTemplateManifestNotFound") })}`
+				);
+
+				return;
+			}
```

Comment:
Let's display specific error message, so the user will be able to understand what failed better.

Reply:
changed